docs/content/languages/go.md

```diff
@@ -0,0 +1,477 @@
+weight: 10
+---
+
+With Golang plugin mechanism, fission supports Golang as one of function languages.
+
+In this usage guide we'll cover how to use this environment, write functions, and work with dependencies.
+
+## Before you start
+
+We'll assume you have Fission and Kubernetes setup.  If not, head over
+to the [install guide](../../installation/).  Verify your Fission setup with:
+
+``` bash
+$ fission --version
+```
+
+## Add the Golang environment to your cluster
+
+Unlike Python, Golang is a compiled language that means we need to compile source code before running it.
+
+Fortunately, builder manager within fission does all these hard works automatically when a Golang function/package is created.
+The Golang builder will convert a source package into a deployable package. 
+
+To enable Golang builder, we need to assign a function builder with flag `--builder`. 
+
+```
+$ fission environment create --name go --image fission/go-env --builder fission/go-builder
+```
+
+## Write a simple function in Golang
+
+Here is a hello world example (hw.go) in Golang:
+
+```go
+// Due to golang plugin mechanism,
+// the package of function handler must be main package
+package main
+
+import (
+    "net/http"
+)
+
+// Handler is the entry point for this fission function
+func Handler(w http.ResponseWriter, r *http.Request) {
+    msg := "Hello, world!\n"
+    w.Write([]byte(msg))
+}
+```
+
+The entrypoint of function let go server to know how to load-in plugin file correctly.
+Here, our first hello world function's entrypoint is simply the name of function: `Handler`.
+
+```bash
+# Create golang env with builder image to build go plugin
+$ fission fn create --name helloworld --env go --src hw.go --entrypoint Handler
+```
+
+Before accessing function, need to ensure deploy package of function is in _succeeded_ state. 
+
+```bash
+$ fission pkg info --name <pkg-name>
+```
+
+Now, let's test our first Golang function with `test` command
+
+```bash
+$ fission fn test --name <function-name>
+```
+
+{{% notice tip %}} 
+See [here](../../usage/trigger) for how to setup different trigger for Golang function. 
+{{% /notice %}} 
+
+## HTTP requests and HTTP responses
+
+From the sample above we know that go server passes HTTP `Request` and `ResponseWriter` to user function for
+further processing.
+
+``` go
+func Handler(w http.ResponseWriter, r *http.Request) {
+    ...
+}
+```
+
+Following we will go through steps for how to accessing/controlling the requests/responses. 
+
+### Accessing HTTP Requests
+
+#### Headers
+
+```go
+package main
+
+import (
+    "fmt"
+    "net/http"
+    "net/textproto"
+)
+
+func Handler(w http.ResponseWriter, r *http.Request) {
+
+    // recommend
+    v1 := r.Header.Get("HEADER_KEY_1")
+
+    // need to convert string cases manually
+    key := textproto.CanonicalMIMEHeaderKey("HEADER_KEY_2")
+    // v2 type is []string
+    v2, _ := r.Header[key]
+
+    w.Write([]byte(fmt.Sprintf("v1: %s, v2: %v", v1, v2)))
+}
+```
+
```

Comment:
can you add a little section here about "create a route using fission route create..."